pkg/controller/migmigration/registry.go

```diff
@@ -23,14 +23,18 @@ func (t *Task) getAnnotations(client k8sclient.Client) (map[string]string, error
 	if registryService == nil {
 		return nil, errors.New("migration registry service not found")
 	}
-	registryDC, err := t.PlanResources.MigPlan.GetRegistryDC(client)
+	registryDC, err := t.PlanResources.MigPlan.GetRegistryDeployment(client)
 	if err != nil {
 		return nil, err
 	}
 	if registryDC == nil {
 		return nil, errors.New("migration registry DeploymentConfig not found")
 	}
 
+	if registryDC.DeletionTimestamp != nil {
+		return nil, errors.New(fmt.Sprintf("DeploymentConfig %s/%s is being garbage collected with deletion timestamp %s", registryDC.Namespace, registryDC.Name, registryDC.DeletionTimestamp))
```

Comment:
> Is this meant to return an error? What is the consequence to us if a DeletionTimestamp is set but has not yet been acted upon?

When the user has deleted the resource but its awaiting garbage collection from kube, it will have a DeletionTimestamp set, its just to safe guard us against an edge case

Reply:
will address other comments in this thread